### PR TITLE
Ntd1hc/feat/automation shall update pypi

### PR DESCRIPTION
**Pull request by ngoan1608**
Hi Thomas,

This PR contains the implementation for ticket #84 .
- Add new job `tag-to-publish-packges-to-pypi` to pipeline to tag repos for publish to pypi (Run parallelly with "Release AIO" job - after successful test jobs )
- [tag_repos_with_their_version_file.sh](https://github.com/test-fullautomation/RobotFramework_AIO/compare/ntd1hc/feat/automation-shall-update-pypi?expand=1#diff-bad39a16778b0abc94effcb430223a945ead70ae3f2f81769039692adecb015e) will get version information from `version.py` file  under given repo (clone repo or update the existing one) then using `git-tag.py` tool to tag repo
- [config/repositories/publish_pypi_repos.txt](https://github.com/test-fullautomation/RobotFramework_AIO/compare/ntd1hc/feat/automation-shall-update-pypi?expand=1#diff-161bf4b9cce37a7eeed0bed1dad8ee3b8f8904b571dbe032083b6566e2a3aece) is configuration file for repos which will be tag for publishing to pypi.

The publish PyPI tag(s) `rel/x.x.x` which is created by this build pipeline will point to the same commit of release AIO tag `rel/aio/y.y.y.y` on each repo.

I have just verified with my personal repos.
You can check with `dev/aio/x.x.x.x` after the test jobs are fixed (they are still failed now so the new job `tag-to-publish-packges-to-pypi` will be skipped)

Thank you,
Ngoan

### CODE REVIEW

File: tools/git-tag/tag_repos_with_their_version_file.sh [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+   # Get version information from version.py file under repo folder
+   version_file=$(ls $version_file_pattern 2>/dev/null)
+   # Check if the version.py does not exist
+   if [ -z "$version_file" ]; then
+      errormsg "No version.py file is found under '${repo_location}'"
+   else
+      echo "Found version file at '${version_file}'"
+   fi
+
+   version_info=$(grep -E 'VERSION\s+=' ${version_file} | awk -F'"' '{print $2}')
```
Comment (test-fullautomation) [inline tools/git-tag/tag_repos_with_their_version_file.sh:70]: Hi Ngoan,
what about to access the version number with a python oneliner? Can be a little safer?
Thank you,
 Thomas

**Review by test-fullautomation — APPROVED** [review]
Hi Ngoan,
thank you for your pull-request.
Please read my comment.
Thank you,
 Thomas

File: tools/git-tag/tag_repos_with_their_version_file.sh (hunk 1 above)
Comment (ngoan1608) [reply tools/git-tag/tag_repos_with_their_version_file.sh:70]: Hi Thomas,

From the tool's view, there is no information about the package name (as sub folder e.g **PyTestLog2DB**) of given repo to be imported then used with python to get the **VERSION** info. I think it is not good to also provide the package name in configuration file.

That is the reason why I use glob pattern to find the certian `version.py` file then to get **VERSION** info from it.

Please let me know if I misunderstood your question.

Thank you,
Ngoan

Comment (test-fullautomation) [reply tools/git-tag/tag_repos_with_their_version_file.sh:70]: Hi Ngoan,
I thought about something like this:
```
C:\A\python-genpackagedoc>"C:\Program Files\RobotFramework\python39\python" -c "import GenPackageDoc.version as gpd;print(gpd.VERSION)"

0.39.0

C:\A\python-genpackagedoc>
```
Thank you,
 Thomas

Comment (ngoan1608) [reply tools/git-tag/tag_repos_with_their_version_file.sh:70]: Hi Thomas,

 Thank your for your proposal!

**GenPackageDoc** is a package name and tool does not have information about it (only repo name `python-genpackagedoc` is given in config file). The current implementation is independent with the package name (no issue when changing package name and maintain only repo name when add new repo for publishing to pypi).

We can use your proposal by adding the package name information besides the repo name in configuration file. Do we need to extend the current implementation of configuration file to add that information?

Thank you,
Ngoan

**Pull request merged**